### PR TITLE
Store: Add a proper error message and UI when the server returns no label rates

### PR DESCRIPTION
**Pull request by DanReyLop**
In the `Label Purchase` flow, when the merchant requested label rates correctly but no rates were returned (an "empty list" response is not the same as an "error" response), the UI would look like this:
image37

This is specially bad with the International Labels feature because it's easy to put items into a box that's not compatible with USPS international (such as `Regional A/B` boxes). The best solution is to not allow the merchant to select those boxes, but that will require at least server changes. The next best solution is to improve the UX.

This is an example of an international shipment. The first package is a `Regional Rate` box (not valid), and the second package is just the item in its original packaging (valid):
screen shot 2018-08-31 at 19 02 23

To test:
* You can test this in Calypso or in WP-Admin (by `npm link`ing this branch into `woocommerce-services`).
* Try the normal case (domestic, or international with a supported package type like a custom box) and see the UI keeps working as usual.
* Force one package to be "invalid" so it doesn't return rates. The easiest way to do it is to select a `Regional Rate Box` in an international shipment.
* Check that the error messages are satisfactory, that the empty `Dropdown` is disabled, and that you can't purchase labels until you change to a correct package.

Marking as `In Progress` while I fix the tests.

### CODE REVIEW

**Comment by jeffstieler** [comment]
I've yet to review the code and test, but I'd like to suggest a copy change:

> No rates available, please double check dimensions and weight or try using different packaging.

instead of

> No rates available, please try using a different box.

File: client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js
Comment (v18) [reply client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js]: Can you elaborate a bit? Where are we handling multiple server errors?

**Comment by v18** [comment]
> No rates were available. Please double check dimensions and weight or try using different packaging.

Can we be a bit more specific about a dimensions or weight problem? I think we already know that if a package is above a certain weight, it can't be shipped with a service - we can mention this. What about dimensions - is there documentation about what's allowed?

File: client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js
Comment (DanReyLop) [reply client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js]: Here: https://github.com/Automattic/wp-calypso/pull/26979/files#diff-8f6bc9adbfa669876e5a156a4c4411cdL86

We were displaying all the errors returned from the server for each package. I removed it because it seemed like overkill, but it may have a reason to be here.

**Comment by DanReyLop** [comment]
> Can we be a bit more specific about a dimensions or weight problem? I think we already know that if a package is above a certain weight, it can't be shipped with a service - we can mention this.

That's not trivial to know for all cases, specially with international shipping. We would need to replicate [this table](https://pe.usps.com/text/imm/immpg.htm).

> What about dimensions - is there documentation about what's allowed?

We have logic for that already in the server: https://github.com/Automattic/woocommerce-connect-server/blob/0824064d6c4913cb94d64c60744e8ec15c877113/lib/shipping/usps/service-data.js#L4-L31

But we aren't sending those limits to the client for client-side validation (yet).

**Comment by DanReyLop** [comment]
Tests are passing now, this is ready for review.

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/test/list.js
Comment (DanReyLop) [inline client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/test/list.js]: The `<ShippingRates/>` component has been vastly simplified in this PR. It no longer renders anything conditionally. Do you folks think it's still worth it to keep some of these tests? For what I've seen, they mostly tested the logic for higind the dropdowns on empty rates, displaying multiple errors, etc (which is no longer relevant).

File: client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js
Comment (jeffstieler) [reply client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js]: AFAIK this logic is to handle errors coming from multiple rates requests per package (one for each carrier in our current architecture). See commit `795cbca1e181476c8f3bde75e782f0828c5efe45` in the server repo.

Given this, I think we should keep the client-side logic until the server-side changes.

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/test/list.js
Comment (jeffstieler) [reply client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/test/list.js]: I'll wait to make a final call to see how this may affect things: https://github.com/Automattic/wp-calypso/pull/26979/files/1518bba22db984af895c400d6112046e9abe45e3#r215014901

In general though, I'm not a big fan of deleting tests unless all the code it tested is also being deleted. How hard is it to adapt the tests to the new behavior?

File: client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js
Comment (DanReyLop) [inline client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js]: Unrelated, but I discovered it while testing another PR. You can check in https://regex101.com/, before you could write anything after a valid ITN (for example, `AES X12345678901234aaaaaaaaaaa!`), now it validates correctly.

Comment (jeffstieler) [reply client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js]: Since we're making changes to the regex - should we also change the capturing groups to be non-capturing? `/^(?:(?:AES X\d{14})|(?:NOEEI 30\.\d{1,2}(?:\([a-z]\)(?:\(\d\))?)?))$/` ?

File: client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js
Comment (jeffstieler) [reply client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js]: I'm good with the test being removed after re-reading. 👍

**Review by jeffstieler — APPROVED** [review]
Code looks good and tests well with domestic and international invalid packaging. (Used 71lbs for domestic, and Regional Rate Box A1 for intl)

Approving with or without the non-capturing regex changes.

**Pull request merged**
